src/Uno.UI/UI/Xaml/Controls/MediaPlayerElement/MediaPlayerPresenter.cs

```diff
@@ -35,7 +35,11 @@ internal void SetOwner(MediaPlayerElement owner)
 			}
 			else
 			{
-				return (float)naturalOtherDimension * scaledOneDimension / (float)naturalOneDimension;
+				// There are situations where between measurements scaledDimension and naturalDimension
+				// have a small difference in value (a few pixels) causing the measure to go into an infinite loop.
+				// Related to: https://github.com/unoplatform/uno/issues/15254
+				var ratio = (float)Math.Round(scaledOneDimension / naturalOneDimension, 0);
```

Comment:
@ajpinedam This would make it for only integer ratios, making the very common 16/9 being 2 as a result. Could you try with 1 or 2 digits, so it stays close to the original ratio?

Reply:
@jeromelaban tried with 2, and there was still a problem, but my tests with 1 passed. Will push an update using a round to 1 decimal.